pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -250,6 +250,9 @@ private BaseFilterOperator constructPhysicalOperator(FilterContext filter, int n
               return new TextContainsFilterOperator(textIndexReader, (TextContainsPredicate) predicate, numDocs);
             case TEXT_MATCH:
               textIndexReader = dataSource.getTextIndex();
+              Preconditions
+                  .checkState(textIndexReader != null, "Cannot apply TEXT_MATCH on column: %s without text index",
+                      column);
```

Comment:
getting this error on ExplainPlanQueriesTest
```
15:52:10.793 ServerQueryExecutorV1Impl - Exception processing requestId 0
java.lang.IllegalStateException: Cannot apply TEXT_MATCH on column: textIndexCol1 without text index
	at com.google.common.base.Preconditions.checkState(Preconditions.java:518) ~[guava-20.0.jar:?]
	at org.apache.pinot.core.plan.FilterPlanNode.constructPhysicalOperator(FilterPlanNode.java:254) ~[classes/:?]
	at org.apache.pinot.core.plan.FilterPlanNode.run(FilterPlanNode.java:92) ~[classes/:?]
```

Reply:
Bug dependency lol